make_image.sh

```diff
@@ -0,0 +1,231 @@
+			usage
+			;;
+		f)
+			FORCE=1
+			;;
+		n)
+			TRUNCATE=0
+			;;
+        t)
+            TOOLCHAIN_DIR=${OPTARG}
+            ;;
+        i)
+			BAREBOX_IMG=${OPTARG}
+			;;
+		a)
+			ERLANG_APP_DIR=${OPTARG}
+			;;
+		n)
+			ERLANG_APP_NAME=${OPTARG}
+			;;
+		v)
+			ERLANG_APP_VSN=${OPTARG}
+			;;
+		l)
+			LOOPBACK_ID=${OPTARG}
+			;;
+		o)
+			OUTPUT_FILE=${OPTARG}
+			;;
+        *)
+            usage 1
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+OUTPUT_FILE="$( cd $( dirname "$OUTPUT_FILE" ); pwd )/$( basename "$OUTPUT_FILE" )"
+
+if [[ "$BAREBOX_IMG" == "" ]]; then
+	BAREBOX_IMG="$( cd "$TOOLCHAIN_DIR/$BAREBOX_RELDIR"; pwd )/$BAREBOX_IMG_FILENAME"
+fi
+
+if [ ! -d "$TOOLCHAIN_DIR" ]; then
+	error 1 "GRiSP 2 toolchain not found at $TOOLCHAIN_DIR"
+fi
+
+if [ ! -f "$BAREBOX_IMG" ]; then
+	error 1 "GRiSP 2 bootloader not found at $BAREBOX_IMG"
+fi
+
+if [ ! -d $(dirname "$OUTPUT_FILE") ]; then
+	error 1 "Ouput file directory not found: $OUTPUT_FILE"
+fi
+
+if [ ! -d "$ERLANG_APP_DIR" ]; then
+	error 1 "Erlang application not found: $ERLANG_APP_DIR"
+fi
+
+if [[ $FORCE == 1 ]]; then
+	rm -f "$OUTPUT_FILE"
+else
+	if [ -f "$OUTPUT_FILE" ]; then
+		error 1 "Output file already exists: $OUTPUT_FILE"
+	fi
+fi
+
+LOOPBACK_DEV="/dev/loop$LOOPBACK_ID"
+LOOPBACK_PART1="${LOOPBACK_DEV}p1"
+LOOPBACK_PART2="${LOOPBACK_DEV}p2"
+TEMPDIR=$( mktemp -d -p "$PWD" )
+TEMPFILE1=$( mktemp -p "$PWD" )
+TEMPFILE2=$( mktemp -p "$PWD" )
+trap onexit 1 2 3 15 ERR
+
+function onexit() {
+	echo "*** CLEANING UP..."
+	if [ -d "$TEMPDIR" ]; then
+		sudo umount "$TEMPDIR"
+		sudo rmdir "$TEMPDIR"
+	fi
+	if [ -e "$LOOPBACK_DEV" ]; then
+		sudo losetup -d "$LOOPBACK_DEV"
+	fi
+	if [ -f "$TEMPFILE1" ]; then
+		sudo rm -f "$TEMPFILE1"
+	fi
+	if [ -f "$TEMPFILE2" ]; then
+		sudo rm -f "$TEMPFILE2"
+	fi
+	exit 1
+}
+
+echo "**************************************************"
+echo "*** BOOTLOADER: $BAREBOX_IMG"
+echo "*** ERLANG APP: $ERLANG_APP_DIR"
+echo "**************************************************"
+
+echo "*** CREATING BASE IMAGE FILE..."
+dd if=/dev/zero of="$TEMPFILE1" bs=1M count=516
+echo "*** WRITING BOOTLOADER..."
+dd if="$BAREBOX_IMG" of="$TEMPFILE1" conv=notrunc
+echo "*** WRITING PARTITION TABLE..."
+sudo modprobe loop
+sudo losetup "$LOOPBACK_DEV" "$TEMPFILE1"
+sudo sfdisk "$LOOPBACK_DEV" << EOF
+unit: sectors
+
+$LOOPBACK_PART1 : start=8192, size=524288, Id=83
+$LOOPBACK_PART2 : start=532480, size=524288, Id=83
+EOF
+echo "*** FORMATING SYSTEM PARTITION A..."
+sudo partprobe "$LOOPBACK_DEV"
+sudo mkfs.vfat -n "GRISP2A" -s 8 "$LOOPBACK_PART1"
+echo "*** MOUNTING SYSTEM PARTITION A..."
+sudo mount -o uid="$USER" "$LOOPBACK_PART1" "$TEMPDIR"
+
+cd "$ERLANG_APP_DIR"
+echo "*** COMPILING ERLANG APPLICATION..."
+rebar3 compile
+echo "*** DEPLOYING ERLANG APPLICATION TO SYSTEM PARTITION A..."
+rebar3 grisp deploy --destination="$TEMPDIR" --pre-script="true" --post-script="true" --force --relname="$ERLANG_APP_NAME" --relvsn="$ERLANG_APP_VSN"
```

Comment:
Do we need `--force` here? The partition should be empty by default, no?

Reply:
I can remove it